exec/cnex/disassembly.c

```diff
@@ -0,0 +1,854 @@
+        case LTS:     disasm_LTS(disasm); break;
+        case GTS:     disasm_GTS(disasm); break;
+        case LES:     disasm_LES(disasm); break;
+        case GES:     disasm_GES(disasm); break;
+        case EQY:     disasm_EQY(disasm); break;
+        case NEY:     disasm_NEY(disasm); break;
+        case LTY:     disasm_LTY(disasm); break;
+        case GTY:     disasm_GTY(disasm); break;
+        case LEY:     disasm_LEY(disasm); break;
+        case GEY:     disasm_GEY(disasm); break;
+        case EQA:     disasm_EQA(disasm); break;
+        case NEA:     disasm_NEA(disasm); break;
+        case EQD:     disasm_EQD(disasm); break;
+        case NED:     disasm_NED(disasm); break;
+        case EQP:     disasm_EQP(disasm); break;
+        case NEP:     disasm_NEP(disasm); break;
+        case EQV:     disasm_EQV(disasm); break;
+        case NEV:     disasm_NEV(disasm); break;
+        case ANDB:    disasm_ANDB(disasm); break;
+        case ORB:     disasm_ORB(disasm); break;
+        case NOTB:    disasm_NOTB(disasm); break;
+        case INDEXAR: disasm_INDEXAR(disasm); break;
+        case INDEXAW: disasm_INDEXAW(disasm); break;
+        case INDEXAV: disasm_INDEXAV(disasm); break;
+        case INDEXAN: disasm_INDEXAN(disasm); break;
+        case INDEXDR: disasm_INDEXDR(disasm); break;
+        case INDEXDW: disasm_INDEXDW(disasm); break;
+        case INDEXDV: disasm_INDEXDV(disasm); break;
+        case INA:     disasm_INA(disasm); break;
+        case IND:     disasm_IND(disasm); break;
+        case CALLP:   disasm_CALLP(disasm); break;
+        case CALLF:   disasm_CALLF(disasm); break;
+        case CALLMF:  disasm_CALLMF(disasm); break;
+        case CALLI:   disasm_CALLI(disasm); break;
+        case JUMP:    disasm_JUMP(disasm); break;
+        case JF:      disasm_JF(disasm); break;
+        case JT:      disasm_JT(disasm); break;
+        case JFCHAIN: disasm_JFCHAIN(disasm); break;
+        case DUP:     disasm_DUP(disasm); break;
+        case DUPX1:   disasm_DUPX1(disasm); break;
+        case DROP:    disasm_DROP(disasm); break;
+        case RET:     disasm_RET(disasm); break;
+        case CONSA:   disasm_CONSA(disasm); break;
+        case CONSD:   disasm_CONSD(disasm); break;
+        case EXCEPT:  disasm_EXCEPT(disasm); break;
+        case ALLOC:   disasm_ALLOC(disasm); break;
+        case PUSHNIL: disasm_PUSHNIL(disasm); break;
+        case RESETC:  disasm_RESETC(disasm); break;
+        case PUSHPEG: disasm_PUSHPEG(disasm); break;
+        case JUMPTBL: disasm_JUMPTBL(disasm); break;
+        case CALLX:   disasm_CALLX(disasm); break;
+        case SWAP:    disasm_SWAP(disasm); break;
+        case DROPN:   disasm_DROPN(disasm); break;
+        case PUSHFP:  disasm_PUSHFP(disasm); break;
+        case CALLV:   disasm_CALLV(disasm); break;
+        case PUSHCI:  disasm_PUSHCI(disasm); break;
+        default:
+            sprintf(disasm->out, "Unknown opcode: %d", disasm->obj->code[disasm->index]);
```

Comment:
And now you're missing the index increment, which means you're back in infinite loop land. You have to increment the index *after* printing the opcode at that index.